lib/utils.py

```diff
@@ -78,8 +79,9 @@ def get_node_ip():
         that is being used as the default gateway.
         This should work on both MacOS X and Linux."""
         try:
-            default_interface = grep(netstat('-nr'), '-e', '^default', '-e' '^0.0.0.0').split()[-1]
-            my_ip = ifaddresses(default_interface)[2][0]['addr']
+            interface_id = next(iter([gateways()][0]['default']), None)
+            default_interface = [gateways()][0]['default'][interface_id][1]
+            my_ip = ifaddresses(default_interface)[min(ifaddresses(default_interface))][0]['addr']
```

Comment:
If i'm not mistaken this is not correct. We cannot just take a min value. Please read paragraph 2 (How do I use it?), pay attention to the use of AF_INET and AF_INET6. If I understand correctly there may be cases when AF_LINK < AF_INET6

Reply:
AF_LINK is always < AF_INET6, which is why I mentioned that its not my preferred solution as it would mean that IPv4 will be selected over IPv6 when both are available.

Other solutions that I came up with would have required a try/except again, which was not wanted above.

Reply:
I'm talking about AF_LINK, not about AF_INET

Reply:
I misunderstood you then. I tested it on multiple raspberries and a few laptops and have not found AF_LINK to be smaller than AF_INET6, but if this solution is not wanted, I currently dont see another without try/except, other than displaying all ips available to the system, which would require larger changes at different places from a quick look I took.